### PR TITLE
Decompile misc battle map funcs

### PR DESCRIPTION
**Pull request by Brotenko**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 13 functions (+0.05%) on `us`.

⚠️ This PR introduces 5 warnings:
- src/battle/area_trd_part_2/4B11F0.c:18: warning: implicit declaration of function `bgm_set_battle_song'
- src/battle/area_trd_part_2/4B11F0.c:19: warning: implicit declaration of function `bgm_push_battle_song'

File: src/battle/area_trd_part_2/4B2550.c [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+#ifdef NON_MATCHING
+#define NAMESPACE dup4_b_area_trd_part_2
+#include "common/UnkWhirlwindEffectFunc.inc.c"
+#define NAMESPACE b_area_trd_part_2
+#else
 INCLUDE_ASM(s32, "battle/area_trd_part_2/4B2550", func_802193B8_4B25A8);
+#endif
 
-INCLUDE_ASM(s32, "battle/area_trd_part_2/4B2550", func_80219490_4B2680);
+//INCLUDE_ASM(s32, "battle/area_trd_part_2/4B2550", dup4_b_area_trd_part_2_UnkEnemyFunc);
```
Comment (ethteck) [inline src/battle/area_trd_part_2/4B2550.c:19]: this comment can go bye bye

File: src/common/UnkWhirlwindEffectFunc.inc.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    s32 var1 = evt_get_variable(script, *args++);
+    s32 var2 = evt_get_variable(script, *args++);
+    s32 var3 = evt_get_variable(script, *args++);
+    s32 var4 = evt_get_variable(script, *args++);
+
+    effectStruct.unk_28 = var1;
+    effectStruct.unk_2C = var2 - 10.0f;
+    effectStruct.unk_30 = var3;
+    
+    playFX_46_whirlwind(6, (s32) &effectStruct, 1.0f, var4);
```
Comment (ethteck) [inline src/common/UnkWhirlwindEffectFunc.inc.c:24]: Could you rename this struct to `EffectWhirlwind`? The struct should go into effects.h and the signature of playFX_46_whirlwind should be changed to take a `EffectWhirlwind*`

**Review by ethteck — CHANGES_REQUESTED** [review]

**Review by ethteck — APPROVED** [review]

**Review by nanaian — APPROVED** [review]
Here's betting UnkEnemyFunc is used for finding this first of the Koopa Bros. that is not on the floor

**Comment by Brotenko** [comment]
> 
> 
> Here's betting UnkEnemyFunc is used for finding the first of the Koopa Bros. that is not on the floor

Well, it has something to do with the Koopa Bros. for sure, that much is certain

**Pull request merged**